butterfly/QueryLayer/DataQuery.py

```diff
@@ -94,3 +94,11 @@ def all_in_some(self, t_index):
         all_in = self.target_bounds - tile_origin
         return np.clip(all_in, 0, self.blocksize)
 
+    def update_source(self, block, dtype):
+        self.RUNTIME.IMAGE.BLOCK.VALUE = block
+        self.OUTPUT.INFO.TYPE.VALUE = dtype
+        self.log('update',
+            path = self.OUTPUT.INFO.PATH.VALUE,
+            block =  block,
+            dtype = dtype)
+
```

Comment:
`DataQuery.update_source` stores and logs the values from `TileQuery.preload_source`, which itself basically just returns results of a call to `DataSource.preload_source`.